internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package variables
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/pointer"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+func Test_ValidateClusterClassVariables(t *testing.T) {
```

Comment:
Can you please add 3 test cases to capture this logic? (empty / null / invalidType)

```go
	switch {
	case len(apiExtensionsSchema.Type) == 0:
		return field.ErrorList{field.Required(fldPath.Child("type"), "type cannot be empty")}
	case apiExtensionsSchema.Type == "null":
		return field.ErrorList{field.Forbidden(fldPath.Child("type"), "type cannot be set to null, use nullable as an alternative")}
	case !validVariableTypes.Has(apiExtensionsSchema.Type):
		return field.ErrorList{field.NotSupported(fldPath.Child("type"), apiExtensionsSchema.Type, validVariableTypes.List())}
	}
```